cohttp-lwt-unix/src/io.ml

```diff
@@ -80,3 +82,25 @@ let catch f =
     | ex -> Lwt.fail ex)
 
 let pp_error = Fmt.exn
+
+let wait_eof_or_closed conn ic =
+  let fd =
+    match conn with
+    | Conduit_lwt_unix.TCP { fd; _ } -> fd
+    | Domain_socket { fd; _ } -> fd
+    | Vchan _ -> assert false
```

Comment:
Do we really want to keep that? I'm not aware of how Vchans are working but, at least, this should be no-op in case of Vchans.